internal/fields/mappings.go

```diff
@@ -0,0 +1,577 @@
+					// TODO: Should this be raised as an error instead?
+					logger.Debugf("field %q is an empty object and it does not exist in the preview", currentPath)
+
+					continue
+				}
+
+				logger.Debugf("Calculating flatten fields for %s", currentPath)
+				flattenFields, err := flattenMappings(currentPath, childField)
+				if err != nil {
+					errs = append(errs, err)
+					return errs
+				}
+
+				for fieldPath, object := range flattenFields {
+					logger.Debugf("- %s", fieldPath)
+
+					def, ok := object.(mappingDefinitions)
+					if !ok {
+						errs = append(errs, fmt.Errorf("invalid field definition/mapping for path: %q", fieldPath))
+						continue
+					}
+
+					if isLocalFieldTypeArray(fieldPath, localSchema) {
+						logger.Debugf("Found field definition with type array, skipping path: %q", fieldPath)
+						continue
+					}
+
+					// TODO: validate mapping with dynamic templates first than validating with ECS
+					// just raise an error if both validation processes fail
+
+					// are all fields under this key defined in ECS
+					err = validateMappingInECS(fieldPath, def, ecsSchema)
+					if err != nil {
+						logger.Warnf("missing key %q in path %q (pending to check dynamic templates)", key, path)
+						errs = append(errs, fmt.Errorf("field %q is undefined: %w", fieldPath, err))
+					}
+				}
+			}
+
+			continue
+		}
+
+		fieldErrs := validateFieldMapping(preview, key, value, currentPath, ecsSchema, localSchema)
+		errs = append(errs, fieldErrs...)
+	}
+	if len(errs) == 0 {
+		return nil
+	}
+	return errs
+}
+
+func validateFieldMapping(preview mappingDefinitions, key string, value any, currentPath string, ecsSchema, localSchema []FieldDefinition) multierror.Error {
+	var errs multierror.Error
+	previewValue := preview[key]
+	switch value.(type) {
+	case map[string]any:
+		// validate field
+		previewField, ok := previewValue.(map[string]any)
+		if !ok {
+			errs = append(errs, fmt.Errorf("unexpected type in preview mappings for path: %q", currentPath))
+		}
+		actualField, ok := value.(map[string]any)
+		if !ok {
+			errs = append(errs, fmt.Errorf("unexpected type in actual mappings for path: %q", currentPath))
+		}
+		logger.Debugf(">>>> Comparing Mappings map[string]any: path %s", currentPath)
+		errs = append(errs, compareMappings(currentPath, mappingDefinitions(previewField), mappingDefinitions(actualField), ecsSchema, localSchema)...)
+	case any:
+		// validate each setting/parameter of the mapping
+		// Skip: mappings should not be able to update, if a mapping exist in both preview and actual, they should be the same.
+
+		// logger.Debugf("Checking mapping Values %s:\nPreview (%T):\n%s\nActual (%T):\n%s\n", currentPath, previewValue, previewValue, value, value)
+		// if previewValue != value {
+		// 	errs = append(errs, fmt.Errorf("unexpected value found in mapping for field %q: preview mappings value (%q) different from the actual mappings value (%q): %q", currentPath, previewValue, value, value))
+		// }
```

Comment:
Currently, it is skipped the comparison of each parameter found in every mapping.